gridscale/resource_gridscale_paas.go

```diff
@@ -288,12 +301,15 @@ func resourceGridscalePaaSServiceRead(d *schema.ResourceData, meta interface{})
 		return fmt.Errorf("%s error setting labels: %v", errorPrefix, err)
 	}
 
-	//Get all available networks
+	// Look for security zone's network that the PaaS service is connected to
+	// (if the paas is connected to security zone. O.w skip)
+	if props.SecurityZoneUUID == "" {
+		return nil
+	}
```

Comment:
why do we need this condition? This breaks the flow for getting the networks! Would u give more info.?

Reply:
That condition is for checking if the paas is using a _security zone_. If so, search for a network that contains that _security zone_, then assign the network's UUID to `network_uuid` (so that the user can use the `network_uuid` to link a VM to the network easier); o.w skip the search. This is the current behaviour of the provider before releasing paas private network.

Reply:
And we've already released private network for paas which means the paas either uses private network or security zone (so `props.SecurityZoneUUID == ""` <=> `props.NetworkUUID != ""`. This condition could save us an API call to fetch all networks and search.